pkg/library/overrides/pods.go

```diff
@@ -72,55 +73,68 @@ func ApplyPodOverrides(workspace *common.DevWorkspaceWithConfig, deployment *app
 	patched.Spec.Template = patchedPodSpecTemplate
 	patched.Spec.Template.Spec.Containers = originalContainers
 	patched.Spec.Template.Spec.InitContainers = originalInitContainers
-	patched.Spec.Template.Spec.Volumes = originalVolumes
 	return patched, nil
 }
 
+func GetPodVolumeOverrides(workspace *dw.DevWorkspaceTemplateSpec) (*[]corev1.Volume, error) {
+	var volumeOverrides []corev1.Volume
+	overrides, err := getPodOverrides(workspace)
+	if err != nil {
+		return &volumeOverrides, err
+	}
+
+	for _, override := range overrides {
+		podSpecTemplate := corev1.PodTemplateSpec{}
+		if err := json.Unmarshal(override.Raw, &podSpecTemplate); err != nil {
+			return &volumeOverrides, fmt.Errorf("error unmarshalling: %w", err)
+		}
+
+		if podSpecTemplate.Spec.Volumes != nil {
+			volumeOverrides = append(volumeOverrides, podSpecTemplate.Spec.Volumes...)
+		}
+	}
+	return &volumeOverrides, nil
```

Comment:
I believe this can return an invalid list of volumes -- what if multiple overrides apply to the same volume?